packages/resource-deployment/scripts/delete-outdated-pools.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# shellcheck disable=SC1090
+set -eo pipefail
+
+# This script will deploy Azure Batch account in user subscription mode
+# and enable managed identity for Azure on Batch pools
+
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -b <batch account name> -r <resource group name> -p <parameter file path> [-t <delete timeout>]
+"
+    exit 1
+}
+
+deletePool() {
+    poolId=$1
+
+    echo "deleting pool $poolId"
+    az batch pool delete --account-name $batchAccountName --pool-id $poolId --yes
+
+    waitForDelete $poolId
+    echo "done"
+}
+
+waitForDelete() {
+    poolId=$1
+
+    checkIfPoolExists $poolId
+    waiting=false
+    end=$((SECONDS + $deleteTimeout))
+    while [ "$poolExists" == "true" ] && [ $SECONDS -le $end ]; do
+        if [ "$waiting" != true ]; then
+            waiting=true
+            echo "Waiting for $poolId to delete"
+            printf " - Running .."
+        fi
+
+        sleep 5
+        printf "."
+        checkIfPoolExists
+    done
+
+    if [ "$poolExists" == "true"]; then
+        echo "Pool did not finish deleting within $deleteTimeout seconds"
+    fi
+}
+
+checkIfPoolExists() {
+    poolId=$1
+
+    poolExists=$(az batch pool list --account-name "$batchAccountName" --query "[?id=='$poolId']" -o tsv)
+    if [[ -z $poolExists ]]; then
+        poolExists=false
+    else
+        poolExists=true
+    fi
+}
+
+deletePoolWhenNodesAreIdle() {
+    poolId=$1
+
+    command="deletePool $poolId"
+    commandName="Delete pool $poolId"
+    . "${0%/*}/run-command-when-batch-nodes-are-idle.sh"
```

Comment:
i would suggest to execute deletion of all pools as a single command to run-command-when-batch-nodes-are-idle, instead of calling run-command-when-batch-nodes-are-idle.sh for each pool.